### ISSUE TITLE
GraphQL - locale: "all" doesn't work with Single Types

### ISSUE THREAD

**Issue by RafidMuhymin** [strapi/strapi#15548]
### Required System information

- Node.js version: 16.18.1
- NPM version: 8.19.2
- Strapi version: 4.5.6
- Database: better-sqlite3@7.6.2
- Operating system: Windows 11

### Describe the bug

HI👋 !  Strapi has recently added support for `locale=all` to REST API for Single Types. And it works flawlessly. But I see that this feature has not been added to GraphQL yet.

### Steps to reproduce the behavior

1. Go to the GraphQL playground
2. Query a single type
3. Add the `locale: "all"`
4. See that instead of treating `all` as all the available locales, it's being treated like a locale code.

### Expected behavior

Like the REST API, an array containing all the localized entries instead of an object.

### Screenshots

<img width="1260" alt="image" src="https://user-images.githubusercontent.com/63650415/214301589-cd0a1d0d-bb80-4b7b-a255-f75758a9325e.png">

### Code snippets

```
{
  taste(locale: "all") {
    data {
      attributes {
        Title 
      }
    }
  }
}
```

### Additional context

N/A

**Comment by petersg83** [comment]
Hello :)
Yes, the issue is the schema. For one locale it's an object and for "all", it would be an array. It's not as easy to implement as in the REST API. Probably some kind of refacto/breaking changes would be involved

**Comment by RafidMuhymin** [comment]
Yeah, I understand that it's not as easy as it sounds. But what kind of breaking changes are you talking about?

**Comment by jastrand** [comment]
Also need this feature for graphql.. Currently this is what I get when I try to query `locale="all"`

` "message": "Undefined binding(s) detected when compiling WHERE. Undefined column(s): [t0.id] query: where \"t0\".\"id\" = ?"`

<img width="1361" alt="image" src="https://user-images.githubusercontent.com/57670121/216972225-155762c4-e50e-4309-a6f6-9c166563b689.png">

**Comment by petersg83** [comment]
GraphQL is very strict, either the schema says the response is an array either it's an object. It cannot be both. So that's why we cannot have sometimes an array and sometimes an object here

**Comment by tomh4** [comment]
Seeing this has been fixed for the REST API, can we expect this to be working in GraphQL as well or should we find some other solution on our side ? :) 
https://github.com/strapi/strapi/issues/12638

**Comment by petersg83** [comment]
From what I understand of the issue it will never be "fixed". If fixed, it would probably be by creating another independent GraphQL Query. So I think using another solution would be best for you anyway :)

**Comment by RafidMuhymin** [comment]
In the latest version `v4.1.1` (in `v4.9.0` too), `locale: "all"` throws an error too.

![image](https://github.com/strapi/strapi/assets/63650415/51fb5271-4716-45dc-97d4-752a16c5cbed)

**Comment by RafidMuhymin** [comment]
The above error actually doesn't happen with every content type. If I create a new minimal content type, I don't get the error.
